### ISSUE TITLE
what is the preprocessing?

### ISSUE THREAD

**Issue by sooyeonshin** [jzbontar/mc-cnn#34]
can i ask you about your work?

I don't understand about preprocessing with kitti data set.

their are several outs exist like x0.bin, x1.bin, metadata...

It's necessary for training? can't be use original image data it self?

**Comment by jzbontar** [comment]
- te: a list of size 40. Contains the indices of the validation set.
- tr: a list of size 154 (194 - 40). Contains the indices of the training set.
- nnz_{tr, te}: Contains the coordinates at which the ground truth is defined (ground truth disparity is not available at all locations of an image). The array is populated in the [make_dataset2](https://github.com/jzbontar/mc-cnn/blob/master/adcensus.cu#L1900) function. 23e6 is just a big number. I know there are not more than 23e6 ground truth disparities (I have an [assert](https://github.com/jzbontar/mc-cnn/blob/master/adcensus.cu#L1917) for that in the C code).

**Comment by Qurey** [comment]
@sooyeonshin @jzbontar 
I don't understand about preprocessing with middlebury data set
in the preprocess_mb.py
` mask = cv2.imread('tmp/mask.png', 0)` 
` disp0[mask != 255] = 0` 
` y, x = np.nonzero(mask == 255).` 
what is the function of mask and what is mask meaning?

**Comment by jzbontar** [comment]
The mask stores information about which areas are occluded. Try opening the generated `mask.png` and it will make sense.
